src/api/auth.rs

```diff
@@ -26,94 +31,157 @@ pub struct NoPromptQuery {
     email: Option<String>,
 }
 
+impl LoginCookie {
+    pub fn removal() -> Cookie<'static> {
+        let mut cookie = Cookie::build(&SETTINGS.auth.login_cookie_name, "")
+            .http_only(true)
+            .path("/")
+            .finish();
+        cookie.make_removal();
+        cookie
+    }
+}
+
+impl TryFrom<Cookie<'static>> for LoginCookie {
+    type Error = ApiError;
+
+    fn try_from(cookie: Cookie<'static>) -> Result<Self, Self::Error> {
+        let mut jar = CookieJar::new();
+        jar.add_original(cookie);
+        match jar
+            .private_mut(&Key::derive_from(&SETTINGS.auth.auth_cookie_key))
+            .get(&SETTINGS.auth.login_cookie_name)
+        {
+            Some(cookie) => {
+                let login_cookie = serde_json::from_str(cookie.value())?;
+                Ok(login_cookie)
+            }
+            None => Err(ApiError::ServerError),
+        }
+    }
+}
+
+impl TryFrom<LoginCookie> for Cookie<'static> {
+    type Error = ApiError;
+
+    fn try_from(login_cookie: LoginCookie) -> Result<Self, Self::Error> {
+        let cookie = Cookie::build(
+            &SETTINGS.auth.login_cookie_name,
+            serde_json::to_string(&login_cookie)?,
+        )
+        .http_only(true)
+        .secure(SETTINGS.auth.auth_cookie_secure)
+        .same_site(SameSite::Lax)
+        .expires(OffsetDateTime::now_utc() + Duration::minutes(15))
```

Comment:
I suppose these two lines answer my previous question of why we're using a seperate login cookie, rather than the session cookie: we'd have to make the session cookie Lax, and we'd have to add a login expiry timestamp to the session then check it.